fix(oauth): instance name encoding on oauth app in admin

byt3sage opened the pull request:
<!--
  - 🚨 SECURITY INFO
  -
  - Before sending a pull request that fixes a security issue please report it via our HackerOne page (https://hackerone.com/nextcloud) following our security policy (https://nextcloud.com/security/). This allows us to coordinate the fix and release without potentially exposing all Nextcloud servers and users in the meantime.
-->

* Resolves: #39956 

## Summary
- This PR aims to resolve the issue with the encoding of the instance name when shown in the OAuth settings in Nextcloud admin.
- This is my first PR for Nextcloud (and hopefully the first of many) so I'm keen for any feedback if this approach doesn't follow what it should.
- I've removed instance name from the translate function. My reasoning for this is the instance name will already be in the users own language and this avoids the issue of the translate function escaping special characters.

## TODO

- [x] Remove instance name binding from the translate function and use string interpolation instead.
- [x] Update translation files to ensure the string prior to the instance name is still translated.

## Screenshots
Before
![image](https://github.com/user-attachments/assets/b1c7fb3d-7cfb-4365-8c70-13f5faba2372)

After
![image](https://github.com/user-attachments/assets/f62abd48-da7c-473d-9f13-e892bb488d02)


## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

icewind1991 committed:
fix: don't recalculate folder size in Cache::delete if the entry didn't exist

Signed-off-by: Robin Appelman <robin@icewind.nl>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

grnd-alt committed:
fix(dav): add activity logging for favorites in dav

Signed-off-by: grnd-alt <salimbelakkaf@outlook.de>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

grnd-alt committed:
refactor(tags): move favorite event dispatching to tags.php

Signed-off-by: grnd-alt <salimbelakkaf@outlook.de>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

come-nc committed:
fix(files_sharing): Do not wrap password policy exception into a generic one

Let the controller access the HintException and show the error to the user.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

grnd-alt committed:
fix(tests): remove redundant unit test for activity

Signed-off-by: grnd-alt <github@belakkaf.net>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

Koc committed:
fix: Fix ServiceWorker scope when Nextcloud installed in subfolder

Signed-off-by: Kostiantyn Miakshyn <molodchick@gmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

nextcloud-command committed:
chore(assets): Recompile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

kesselb committed:
feat(dashboard): wish happy birthday

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

icewind1991 committed:
chore: cleanup leftover debug statement

Signed-off-by: Robin Appelman <robin@icewind.nl>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

come-nc committed:
fix(shares): Wrap exceptions from password validation to set code to 400

This fixes a regression that bad password returned 403 instead of 400
 because of previous changes.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

come-nc committed:
fix(tests): Adapt ManagerTest to change in Exception class used

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

nfebe committed:
fix(files_sharing): Correct property enforced property names

Signed-off-by: nfebe <fenn25.fn@gmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

nextcloud-command committed:
chore(assets): Recompile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

skjnldsv committed:
fix(files): throttle favorite with max 5 simultaneous requests

Signed-off-by: skjnldsv <skjnldsv@protonmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

nextcloud-command committed:
chore(assets): Recompile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

byt3sage committed:
bug: moved instance name out of l10n to fix issue with instance name decoding

Signed-off-by: byt3sage <jjt@tfmail.co.uk>

byt3sage committed:
chore: formatting

Signed-off-by: byt3sage <jjt@tfmail.co.uk>

byt3sage committed:
chore: formatting

Signed-off-by: byt3sage <jjt@tfmail.co.uk>

byt3sage committed:
Merge branch 'master' into bug/oauth-instance-name-decoding-39956

Pytal committed:
feat(files): Right-align list actions

Signed-off-by: Christopher Ng <chrng8@gmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

Pytal committed:
feat(trashbin): Allow emptying trash

Signed-off-by: Christopher Ng <chrng8@gmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

nextcloud-bot committed:
Fix(l10n): Update translations from Transifex

Signed-off-by: Nextcloud bot <bot@nextcloud.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

skjnldsv committed:
fix(files_sharing): also submit new share on password submit

Signed-off-by: skjnldsv <skjnldsv@protonmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

nextcloud-command committed:
chore(assets): Recompile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

skjnldsv committed:
fix(files_trashbin): update @nextcloud/files and adjust api + l10n fix

Signed-off-by: skjnldsv <skjnldsv@protonmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

skjnldsv committed:
fix(files): protect filelist actions with try...catch

Signed-off-by: skjnldsv <skjnldsv@protonmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

skjnldsv committed:
feat(files_trashbin): add cypress tests

Signed-off-by: skjnldsv <skjnldsv@protonmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

skjnldsv committed:
perf(files_trashbin): use empty trashbin endpoint instead of batch delete requests

Signed-off-by: skjnldsv <skjnldsv@protonmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

skjnldsv committed:
feat(files_trashbin): improve wordings

Signed-off-by: skjnldsv <skjnldsv@protonmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

icewind1991 committed:
fix: handle calling select on partitioned queries with an array of columns

Signed-off-by: Robin Appelman <robin@icewind.nl>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

icewind1991 committed:
fix: improve logic for ensuring join columns are selected for partitioned queries

Signed-off-by: Robin Appelman <robin@icewind.nl>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

kesselb commented:
Thanks for your pull request :+1: 

Please revert the changes to the language files. The files are synced automatically from Transifex. It's enough to change the source.

Pytal committed:
fix(trashbin): Fix infinitely loading empty trash action on close

Signed-off-by: Christopher Ng <chrng8@gmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

Pytal committed:
chore: compile assets

Signed-off-by: Christopher Ng <chrng8@gmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

Pytal committed:
chore(psalm): Update baseline

Signed-off-by: Christopher Ng <chrng8@gmail.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

dependabot[bot] committed:
chore(deps): bump nanoid from 3.3.7 to 3.3.8

Bumps [nanoid](https://github.com/ai/nanoid) from 3.3.7 to 3.3.8.
- [Release notes](https://github.com/ai/nanoid/releases)
- [Changelog](https://github.com/ai/nanoid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ai/nanoid/compare/3.3.7...3.3.8)

---
updated-dependencies:
- dependency-name: nanoid
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

nextcloud-command committed:
chore(assets): Recompile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

nextcloud-bot committed:
Fix(l10n): Update translations from Transifex

Signed-off-by: Nextcloud bot <bot@nextcloud.com>
Signed-off-by: byt3sage <jjt@tfmail.co.uk>

byt3sage committed:
chore: reverted language files

byt3sage committed:
chore: reverted language files

Signed-off-by: byt3sage <jjt@tfmail.co.uk>

byt3sage commented:
@kesselb Ah that makes sense! I've reverted these back now.

byt3sage committed:
Merge branch 'master' of github.com:byt3sage/server into bug/oauth-instance-name-decoding-39956

byt3sage committed:
Merge branch 'bug/oauth-instance-name-decoding-39956' of github.com:byt3sage/server into bug/oauth-instance-name-decoding-39956

Signed-off-by: byt3sage <jjt@tfmail.co.uk>

byt3sage committed:
Merge branch 'bug/oauth-instance-name-decoding-39956' of github.com:byt3sage/server into bug/oauth-instance-name-decoding-39956

byt3sage committed:
Merge branch 'master' into bug/oauth-instance-name-decoding-39956

byt3sage committed:
Merge branch 'bug/oauth-instance-name-decoding-39956' of github.com:byt3sage/server into bug/oauth-instance-name-decoding-39956

byt3sage committed:
Merge branch 'bug/oauth-instance-name-decoding-39956' of github.com:byt3sage/server into bug/oauth-instance-name-decoding-39956

byt3sage committed:
Merge branch 'master' into bug/oauth-instance-name-decoding-39956

kesselb commented:
Hi @byt3sage 

Thank you, it appears that the last rebase went wrong. Could you please retry?
